tests/ci/build_run_benchmarks.sh

```diff
@@ -0,0 +1,138 @@
+mkdir "${PR_FOLDER_NAME}"/build
+cmake -B"${PR_FOLDER_NAME}"/build -H"${PR_FOLDER_NAME}" -GNinja -DCMAKE_BUILD_TYPE=Release \
+  -DAWSLC_INSTALL_DIR="${AWSLC_PR_ROOT}" \
+  -DBORINGSSL_INSTALL_DIR="${BORINGSSL_ROOT}" \
+  -DOPENSSL_INSTALL_DIR="${OPENSSL_ROOT}"
+ninja -C "${PR_FOLDER_NAME}"/build
+
+# build FIPS compliant version of AWSLC pr
+mkdir "${PR_FOLDER_NAME}"/fips_build
+cmake -B"${PR_FOLDER_NAME}"/fips_build -H"${PR_FOLDER_NAME}" -GNinja -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -DAWSLC_INSTALL_DIR="${AWSLC_PR_ROOT}"
+ninja -C "${PR_FOLDER_NAME}"/fips_build
+
+# build AWSLC prod
+mkdir aws-lc-prod/build
+cmake -Baws-lc-prod/build -Haws-lc-prod -GNinja -DCMAKE_BUILD_TYPE=Release -DAWSLC_INSTALL_DIR="${AWSLC_PROD_ROOT}"
+ninja -C aws-lc-prod/build
+
+#build FIPS compliant version of AWSLC prod
+mkdir aws-lc-prod/fips_build
+cmake -Baws-lc-prod/fips_build -Haws-lc-prod -GNinja -DFIPS=1 -DCMAKE_BUILD_TYPE=Release -DAWSLC_INSTALL_DIR="${AWSLC_PROD_ROOT}"
+ninja -C aws-lc-prod/fips_build
+
+# run the generated benchmarks and wait for them to finish
+taskset -c 0 ./aws-lc-pr/build/tool/awslc_bm -timeout 3 -json > aws-lc-pr_bm.json &
+pr_pid=$!
+taskset -c 1 ./aws-lc-pr/fips_build/tool/awslc_bm -timeout 3 -json > aws-lc-pr_fips_bm.json &
+pr_fips_pid=$!
+
+taskset -c 2 ./aws-lc-prod/build/tool/awslc_bm -timeout 3 -json > aws-lc-prod_bm.json &
+prod_pid=$!
+taskset -c 3 ./aws-lc-prod/fips_build/tool/awslc_bm -timeout 3 -json > aws-lc-prod_fips_bm.json &
+prod_fips_pid=$!
+
+taskset -c 4 ./aws-lc-pr/build/tool/ossl_bm -timeout 3 -json > ossl_bm.json &
+ossl_pid=$!
+taskset -c 5 ./aws-lc-pr/build/tool/bssl_bm -timeout 3 -json > bssl_bm.json &
+bssl_pid=$!
+
+# wait for benchmarks to finish
+wait "${pr_pid}"
+wait "${pr_fips_pid}"
+wait "${prod_pid}"
+wait "${prod_fips_pid}"
+wait "${ossl_pid}"
+wait "${bssl_pid}"
+
+# convert results from .json to .csv
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py aws-lc-pr_bm.json
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py aws-lc-pr_fips_bm.json
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py aws-lc-prod_bm.json
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py aws-lc-prod_fips_bm.json
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py ossl_bm.json
+python3 aws-lc-pr/tests/ci/benchmark_framework/convert_json_to_csv.py bssl_bm.json
+
+# check for regressions!
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py aws-lc-prod_bm.csv aws-lc-pr_bm.csv prod_vs_pr.csv
+prod_vs_pr_code="$?"
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py aws-lc-prod_fips_bm.csv aws-lc-pr_fips_bm.csv prod_vs_pr_fips.csv
+prod_vs_pr_fips_code="$?"
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py ossl_bm.csv aws-lc-pr_bm.csv ossl_vs_pr.csv
+ossl_vs_pr_code="$?"
+python3 aws-lc-pr/tests/ci/benchmark_framework/compare_results.py bssl_bm.csv aws-lc-pr_bm.csv bssl_vs_pr.csv
```

Comment:
You are looking for regression in the AWS-LC+PR against tip of BSSL.

So, let's say BSSL implements this nice optimisation that is not yet in AWS-LC for some algorithm `A`. This optimisation is neat, reducing cost by 50%!

What happens when I submit a PR against AWS-LC? You will take tip of BSSL, and compare performance of algorithms, including `A`, against the performance of algorithms in AWS-LC+PR. This will produce a regression event.

How do we mitigate this?